### DISCUSSION TITLE
advice needed to use workers in sonification library

### DISCUSSION THREAD

**Discussion by venkateshpotluri** [make4all/psst#85] [Q&A]
I am refactoring the sonifier code to address #60. As mentioned in the issue description, we need to change how we schedule sound to [this approach](https://www.html5rocks.com/en/tutorials/audio/scheduling/). I am implementing a web worker that acts as a timer to schedule sounds as mentioned in the article and am looking for some advice to use web workers in the project setup we have. My code is hear: https://github.com/make4all/sonification/commit/7f0e971ae2d77cb93fb36713c13632c965d07a27
specifically, I get a runtime error as follows: Error
timerWorker.ts:1 Search for this message on the Web Uncaught SyntaxError: Unexpected token '<'

I tried reading up on using web workers in typescript and all the information I find mentions use of react with typescript, and suggests to use webpack and/or eject the react app. My understanding is that the above error is due to my improper use of workers in our project setup. However, I am unsure of the best way to fix this i.e. use web workers in our project setup. Please provide pointers and advice to address this. Tagging @jrthompson33  and @pelikhan  as we discussed this code change and the build system that we use.

**Comment by jrthompson33** [comment]
Hey @venkateshpotluri! I took a look at this branch - there seems to be two issues with the web worker:

1. The `timerWorker.ts` code should be in loaded in the build or placed in the `public` directory (i.e., move the file and change it to be vanilla javascript at \public\timerWorker.js). Changing to javascript is needed because it won't be compiled by webpack.
2. According to the [API on Web Workers](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers) it looks like you cannot access global variables such as `window` or `document` since the web worker is meant to be used for heavy computations.

Having said all of that, I'm not convinced that you need to use a web worker to implement the timer to schedule sounds.
